### PR TITLE
Feat/error alert #43

### LINKED ISSUE #43: Show error alert notice when any API request fails (both map and chart) (2)
Implement general purpose error notice component. Multiple errors can be visible at the same time. They should stack on top of one another, from the top down, with some margin spacing between them.
X closes the error, if X is not clicked within 10 seconds it closes itself automatically.

Use this initially to show an error when any API request fails: "Error fetching data. Retrying now…"

Now that we can show useful error messages, we should also make the map failure state more elegant. When the api call to get the data for the map fails, it should show the new error message, but the map should still be visible (showing the data from the previous API call that succeeded).

image

### PR DESCRIPTION
**Pull request by dlinah**
# Pull Request

## Description

show error alert on request failure 

Fixes #43

## How Has This Been Tested?

- tested locally

## Ui changes
Screenshot 2022-06-20 at 00 28 53

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by dlinah** [comment]
@peterdudfield @dctanner 
this is the apis we call , i added some suggestions , feel free to update them (i've added codes like you suggested peter so we can debug error when it happens)
```

Map Requests:
initial map data: /GB/solar/gsp/forecast/all => "Error fetching initial data. Retrying now…"
/GB/solar/gsp/forecast/all?historic=true&normalize=true => "Error fetching map data. Retrying now…"

National chart Requests:
/GB/solar/gsp/forecast/latest/0 => "Error fetching national forcasts data. Retrying now…"
/GB/solar/gsp/truth/one_gsp/0/?regime=in-day => "Error fetching "NA-INDAY" data. Retrying now…"
/GB/solar/gsp/truth/one_gsp/0/?regime=day-after => "Error fetching "NA-DAYAFTER" data. Retrying now…"

Gsp chart Requests:
/GB/solar/gsp/forecast/all?historic=true&normalize=true
/GB/solar/gsp/truth/one_gsp/${gspId}/?regime=in-day => "Error fetching "GSP-INDAY" data. Retrying now…"
/GB/solar/gsp/truth/one_gsp/${gspId}/?regime=day-after => "Error fetching "GSP-DAYAFTER" data. Retrying now…"

```

also i'm adding send error to sentry in this pr

File: apps/nowcasting-app/constant.ts [hunk 1]
```diff
@@ -6,3 +6,36 @@ export const getAllForecastUrl = (isNormalized: boolean, isHistoric: boolean) =>
...
+    msg: 'Error fetching "NA-INDAY" data. Retrying now…',
+  },
+  {
+    key: /\/GB\/solar\/gsp\/truth\/one_gsp\/0\/\?regime=day-after$/,
+    msg: 'Error fetching "NA-DAYAFTER" data. Retrying now…',
+  },
+
+  {
+    key: /\/GB\/solar\/gsp\/truth\/one_gsp\/\d+\/\?regime=in-day$/,
+    msg: 'Error fetching "GSP-INDAY" data. Retrying now…',
```
Comment (peterdudfield) [inline apps/nowcasting-app/constant.ts:35]: Error fetching GSP {gsp} PV Live initial estimate. Retrying. now ...,

File: apps/nowcasting-app/constant.ts [hunk 2]
```diff
@@ -1,7 +1,47 @@
...
+    key: /\/GB\/solar\/gsp\/forecast\/latest\/0$/,
+    getMsg: (key: string) => "Error fetching national forecasts data. Retrying now…",
+  },
+  {
+    key: /\/GB\/solar\/gsp\/truth\/one_gsp\/0\/\?regime=in-day$/,
+    getMsg: (key: string) => "Error fetching National PV Live initial estimate. Retrying. now ...",
+  },
+  {
+    key: /\/GB\/solar\/gsp\/truth\/one_gsp\/0\/\?regime=day-after$/,
+    getMsg: (key: string) => "Error fetching National PV Live updated. Retrying. now ...",
```
Comment (peterdudfield) [inline apps/nowcasting-app/constant.ts:30]: should it be 'Retrying now ....', i.e remove extra .

Comment (peterdudfield) [reply apps/nowcasting-app/constant.ts:30]: and same below

**Review by Alaatohamy — APPROVED** [review]

**Comment by peterdudfield** [comment]
Looks great, I ran it localy and forced an API error

Screenshot 2022-06-28 at 11 00 12

**Review by peterdudfield — APPROVED** [review]

**Pull request merged**